emasamplerate.go

```diff
@@ -127,18 +128,29 @@ func (e *EMASampleRate) Start() error {
 // updateMaps calculates a new saved rate map based on the contents of the
 // counter map
 func (e *EMASampleRate) updateMaps() {
-	// make a local copy of the sample counters for calculation
 	e.lock.Lock()
-	tmpCounts := e.currentCounts
-	e.currentCounts = make(map[string]float64)
-	e.currentBurstSum = 0
-	e.lock.Unlock()
+	if e.testSignalMapsDone != nil {
+		defer func() {
+			e.testSignalMapsDone <- struct{}{}
```

Comment:
I realize this is just for tests but will it cause an issue that the signal is sent after the lock is released? This is a change from when it was down below and the signal was sent before the `defer` was called that would release the lock.

Reply:
This is OK - it's only there to signal that the function has exited, since it's called asynchronously.